gtk4/src/expression.rs

```diff
@@ -0,0 +1,362 @@
+        ref => |ptr| ffi::gtk_expression_ref(ptr),
+        unref => |ptr| ffi::gtk_expression_unref(ptr),
+    }
+}
+
+impl glib::StaticType for Expression {
+    #[doc(alias = "gtk_expression_get_type")]
+    fn static_type() -> Type {
+        unsafe { from_glib(ffi::gtk_expression_get_type()) }
+    }
+}
+
+impl Expression {
+    #[doc(alias = "gtk_expression_get_value_type")]
+    pub fn get_value_type(&self) -> Type {
+        skip_assert_initialized!();
+        unsafe { from_glib(ffi::gtk_expression_get_value_type(self.to_glib_none().0)) }
+    }
+
+    #[doc(alias = "gtk_expression_is_static")]
+    pub fn is_static(&self) -> bool {
+        skip_assert_initialized!();
+        unsafe { from_glib(ffi::gtk_expression_is_static(self.to_glib_none().0)) }
+    }
+
+    #[doc(alias = "gtk_expression_bind")]
+    pub fn bind<T: IsA<Object>>(
+        &self,
+        target: &T,
+        property_name: &str,
+        this: Option<&T>,
+    ) -> ExpressionWatch {
+        assert_initialized_main_thread!();
+        unsafe {
+            let ret = ffi::gtk_expression_bind(
+                self.to_glib_none().0,
+                target.as_ref().to_glib_none().0,
+                property_name.to_glib_none().0,
+                this.map(|t| t.as_ref()).to_glib_none().0,
+            );
+            ExpressionWatch(*ret)
+        }
+    }
+
+    #[doc(alias = "gtk_expression_evaluate")]
+    pub fn evaluate<T: IsA<Object>>(&self, this: Option<&T>) -> Option<Value> {
+        skip_assert_initialized!();
+        unsafe {
+            let mut value = Value::uninitialized();
+            let ret = ffi::gtk_expression_evaluate(
+                self.to_glib_none().0,
+                this.map(|t| t.as_ref()).to_glib_none().0,
+                value.to_glib_none_mut().0,
+            );
+            if from_glib(ret) {
+                Some(value)
+            } else {
+                None
+            }
+        }
+    }
+
+    #[doc(alias = "gtk_expression_watch")]
+    pub fn watch<T: IsA<Object>, F: Fn() + 'static>(
+        &self,
+        this: Option<&T>,
+        notify: F,
+    ) -> ExpressionWatch {
+        assert_initialized_main_thread!();
+        unsafe extern "C" fn notify_trampoline<F: Fn() + 'static>(user_data: glib::ffi::gpointer) {
+            let f: &F = &*(user_data as *const F);
+            f()
+        }
+        unsafe extern "C" fn destroy_func<F: Fn() + 'static>(user_data: glib::ffi::gpointer) {
+            let _callback: Box_<Option<Box_<F>>> = Box_::from_raw(user_data as *mut _);
+        }
+        let callback_data: Box_<F> = Box_::new(notify);
+        unsafe {
+            let ret = ffi::gtk_expression_watch(
+                self.to_glib_none().0,
+                this.map(|t| t.as_ref()).to_glib_none().0,
+                Some(notify_trampoline::<F> as _),
+                Box_::into_raw(callback_data) as *mut _,
+                Some(destroy_func::<F> as _),
+            );
+            ExpressionWatch(*ret)
```

Comment:
This is wrong

> Note that the only reference held to the watch will be released when the watch is unwatched which can happen automatically, and not just via gtk_expression_watch_unwatch(). You should call gtk_expression_watch_ref() if you want to keep the watch around.

You need to use `from_glib_full()` here or otherwise increase the reference count, or otherwise we'll easily have a dangling pointer.